src/main/java/uk/gov/hmcts/probate/service/exceptionrecord/ExceptionRecordService.java

```diff
@@ -114,4 +140,104 @@ public SuccessfulTransformationResponse createGrantOfRepresentationCaseFromExcep
+                caseReference = ocrFieldValues.get(CAVEAT_EXTEND_CASE_REFERENCE_KEY);
+            }
+
+            Assert.notEmpty(
+                    erRequest.getScannedDocuments(),
+                    "Missing scanned documents in Exception Record"
+            );
+
+            Assert.isTrue(
+                    caveatDetails.getId().toString().equals(caseReference),
+                    "Case retrieved does not match OCR data for caseReference"
+            );
+
+            // Create CaveatCallbackRequest
+            CaveatCallbackRequest caveatCallbackRequest = new CaveatCallbackRequest(caveatDetails);
+
+            // Add scanned documents
+            log.info("Mapping Caveat Scanned Documents to case.");
+            uk.gov.hmcts.probate.model.ccd.caveat.request.CaveatData caveatData = caveatDetails.getData();
+            caveatCallbackRequest.getCaseDetails().getData().setScannedDocuments(
+                    mergeScannedDocuments(caveatData.getScannedDocuments(), erRequest.getScannedDocuments(), erRequest.getId()));
+
+            // Validate caveat extension
+            log.info("Validating caveat extension.");
+            CaveatCallbackResponse caveatCallbackResponse = eventValidationService.validateCaveatRequest(caveatCallbackRequest, validationRuleCaveatsExpiry);
+            if (caveatCallbackResponse.getErrors().isEmpty()) {
+                LocalDate defaultExpiry = caveatCallbackRequest.getCaseDetails().getData()
+                        .getExpiryDate().plusMonths(CaveatCallbackResponseTransformer.CAVEAT_EXPIRY_EXTENSION_PERIOD_IN_MONTHS);
+                log.info("No errors found with validateCaveatRequest, updating expiryDate to {} in request.",
+                        defaultExpiry.format(CaveatCallbackResponseTransformer.dateTimeFormatter));
+                caveatCallbackRequest.getCaseDetails().getData().setExpiryDate(defaultExpiry);
+                log.info("Calling caveatExtend to notify of caveator of extension.");
+                caveatCallbackResponse = caveatNotificationService.caveatExtend(caveatCallbackRequest);
+                log.info("Call to caveatExtend was successful.");
+            }
+
+            return SuccessfulCaveatUpdateResponse.builder()
+                    .caseUpdateDetails(caveatCallbackResponse.getCaveatData())
+                    .warnings(caveatCallbackResponse.getWarnings())
+                    .errors(caveatCallbackResponse.getErrors())
+                    .build();
+
+        } catch(Exception e){
+            log.error("Error Extending Caveat case from Exception Record", e);
+            throw new OCRMappingException(e.getMessage());
+        }
+    }
+
+    private List<uk.gov.hmcts.probate.model.ccd.raw.CollectionMember<uk.gov.hmcts.probate.model.ccd.raw.ScannedDocument>> mergeScannedDocuments(
+            List<uk.gov.hmcts.probate.model.ccd.raw.CollectionMember<uk.gov.hmcts.probate.model.ccd.raw.ScannedDocument>> caseScannedDocuments,
```

Comment:
caseScannedDocuments.size() needs to be compared with newScannedDocuments.size() after the merge. If both are the same then Bulk Scanning have told us they will not update. So I think an OCRValidationExcption needs to be thrown if this is the case with an appropriate message to halt the update.

Reply:
done on my commit - needs checking